pkg/conversion/converter.go

```diff
@@ -670,6 +670,32 @@ func (c *Converter) defaultConvert(sv, dv reflect.Value, scope *scope) error {
 		switch st.Kind() {
 		case reflect.Ptr, reflect.Interface:
 			return c.convert(sv.Elem(), dv.Elem(), scope)
+
+		case reflect.Slice:
+			// url values enter here as string slices.
+			// if the destinationvalue can unmarshal json and the source value is a string slice with exactly one value, unmarshal the string
+			// if this doesn't work for some reason, call the default convert instead of failing
+			if sv.Len() != 1 {
+				return c.convert(sv, dv.Elem(), scope)
+			}
+			sliceElement := sv.Index(0)
+			if sliceElement.Kind() != reflect.String {
+				return c.convert(sv, dv.Elem(), scope)
+			}
+
+			unmarshalFunction := dv.MethodByName("UnmarshalJSON")
```

Comment:
> Is there any way this can result in arbitrary code execution?

If someone writes a bad custom json marshal/umarshal pair, then yes.  But if you've included the type inside of your go code and it does custom marshalling, then you're already exposed to its bugs.  There is no additional risk by doing it for params as well as json.